server/service/apple_mdm_test.go

```diff
@@ -228,3 +228,106 @@ func TestMDMAppleEnrollURL(t *testing.T) {
 		require.Equal(t, tt.expectedURL, enrollURL)
 	}
 }
+
+func TestAppleMDMEnrollmentProfile(t *testing.T) {
+	svc, ctx, _ := setupAppleMDMService(t)
+
+	// Only global admins can create enrollment profiles.
+	ctx = test.UserContext(ctx, test.UserAdmin)
+	_, err := svc.NewMDMAppleEnrollmentProfile(ctx, fleet.MDMAppleEnrollmentProfilePayload{})
+	require.NoError(t, err)
+
+	// All other users should not have access to the endpoints.
+	for _, user := range []*fleet.User{
+		test.UserNoRoles,
+		test.UserMaintainer,
+		test.UserObserver,
+		test.UserTeamAdminTeam1,
+	} {
+		ctx := test.UserContext(ctx, user)
+		_, err := svc.NewMDMAppleEnrollmentProfile(ctx, fleet.MDMAppleEnrollmentProfilePayload{})
+		require.Error(t, err)
+		require.Contains(t, err.Error(), authz.ForbiddenErrorMessage)
+	}
+}
+
+func TestMDMAuthenticate(t *testing.T) {
+	ds := new(mock.Store)
+	svc := MDMAppleCheckinAndCommandService{ds: ds}
+	ctx := context.Background()
+	uuid, serial, model := "ABC-DEF-GHI", "XYZABC", "MacBookPro 16,1"
+
+	ds.IngestMDMAppleDeviceFromCheckinFunc = func(ctx context.Context, mdmHost fleet.MDMAppleHostDetails) error {
+		require.Equal(t, uuid, mdmHost.UDID)
+		require.Equal(t, serial, mdmHost.SerialNumber)
+		require.Equal(t, model, mdmHost.Model)
+		return nil
+	}
+
+	ds.GetHostMDMCheckinInfoFunc = func(ct context.Context, hostUUID string) (*fleet.HostMDMCheckinInfo, error) {
+		require.Equal(t, uuid, hostUUID)
+		return &fleet.HostMDMCheckinInfo{HardwareSerial: serial, InstalledFromDEP: false}, nil
+	}
+
+	ds.NewActivityFunc = func(ctx context.Context, user *fleet.User, activity fleet.ActivityDetails) error {
+		a, ok := activity.(*fleet.ActivityTypeMDMEnrolled)
+		require.True(t, ok)
+		require.Nil(t, user)
+		require.Equal(t, "mdm_enrolled", activity.ActivityName())
+		require.Equal(t, serial, a.HostSerial)
+		require.False(t, a.InstalledFromDEP)
+		return nil
+	}
+
+	err := svc.Authenticate(
+		&mdm.Request{Context: ctx},
+		&mdm.Authenticate{
+			Enrollment: mdm.Enrollment{
+				UDID: uuid,
+			},
+			SerialNumber: serial,
+			Model:        model,
+		},
+	)
+	require.NoError(t, err)
```

Comment:
Just to be safe, you should assert that the mocks have been invoked, e.g.:

```
require.True(t, ds.IngestMDMAppleDeviceFromCheckinFuncInvoked)
```

Reply:
lol, without this the test might very well not exist. Thanks!